matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/failure/Extensions.kt

```diff
@@ -99,6 +100,11 @@ fun Throwable.isInvalidUIAAuth(): Boolean {
             error.flows != null
 }
 
+fun Throwable.isHomeserverUnavailable(): Boolean {
+    return this is Failure.NetworkConnection &&
+            this.ioException is UnknownHostException
+}
+
```

Comment:
Optional: I think the `fun Throwable.isHomeserverUnavailable() = this is...` syntax would be nicer here

Reply:
Thanks for making this change. Looking awesome!